### ISSUE TITLE
[CI] Quarkus main fails with Java 23 mandrel build of `graal/master` on Linux

### ISSUE THREAD

**Issue by zakkak** [graalvm/mandrel#652]
This issue is automatically updated by CI to report issues with Quarkus main and mandrel builds of `graal/master`

**Comment by zakkak** [comment]
Failures are due to https://github.com/adoptium/api.adoptium.net/issues/840

**Comment by jerboaa** [comment]
Build fix in progress in https://github.com/graalvm/mandrel-packaging/pull/384

**Comment by zakkak** [comment]
* Q IT Main issue is https://github.com/quarkusio/quarkus/issues/37809 (fix WIP)
* Q IT Messaging1 issue is https://github.com/quarkusio/quarkus/issues/37862 (fix WIP)
* Q IT Misc2 issue is https://github.com/quarkusio/quarkus/issues/37657 (upstream issue: https://github.com/oracle/graal/issues/8075)

**Comment by jerboaa** [comment]
When generating the native-image-agent the build fails with:

```
        58 types,    58 fields, and    52 methods registered for JNI access
        4 native libraries: dl, pthread, rt, z

Error: Invalid library path /home/runner/work/mandrel/mandrelvm/lib/svm/clibraries/linux-amd64/glibc
```

I'll look into it.

**Comment by jerboaa** [comment]
The full error is:

```
Error: Invalid library path /disk/graal/upstream-sources/graal/mandrel-build/lib/svm/clibraries/linux-amd64/glibc
com.oracle.svm.core.util.UserError$UserException: Invalid library path /disk/graal/upstream-sources/graal/mandrel-build/lib/svm/clibraries/linux-amd64/glibc
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:94)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.getAllStaticLibs(NativeLibraries.java:480)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.getStaticLibraries(NativeLibraries.java:453)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.VMFeature.afterAnalysis(VMFeature.java:102)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$13(NativeImageGenerator.java:823)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:92)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:823)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:571)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:537)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:526)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:701)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:140)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:95)
Caused by: java.nio.file.NoSuchFileException: /disk/graal/upstream-sources/graal/mandrel-build/lib/svm/clibraries/linux-amd64/glibc
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.asIOException(UnixException.java:115)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newDirectoryStream(UnixFileSystemProvider.java:501)
	at java.base/java.nio.file.Files.newDirectoryStream(Files.java:482)
	at java.base/java.nio.file.Files.list(Files.java:3790)
	at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c.NativeLibraries.getAllStaticLibs(NativeLibraries.java:475)
	... 11 more
------------------------------------------------------------------------------------------------------------------------
```

And the path indeed doesn't exist. This is a mandrel-packaging bug.

**Comment by jerboaa** [comment]
> This is a mandrel-packaging bug.

Proposed fix in https://github.com/graalvm/mandrel-packaging/pull/385

**Comment by jerboaa** [comment]
* Q IT Main issue is https://github.com/quarkusio/quarkus/issues/37809

**Comment by jerboaa** [comment]
* Q IT Main issue is https://github.com/quarkusio/quarkus/issues/37809

**Comment by jerboaa** [comment]
* Q IT Main issue is https://github.com/quarkusio/quarkus/issues/37809

**Comment by zakkak** [comment]
Issue tracked in https://github.com/graalvm/mandrel/issues/663

**Comment by jerboaa** [comment]
Issue tracked in https://github.com/graalvm/mandrel/issues/663

**Comment by zakkak** [comment]
* Q IT Main issue is https://github.com/quarkusio/quarkus/issues/37809, fix merged today, will probably go away tomorrow.
* Q IT Data 6 is new. The issue seems irrelevant/intermittent:

```
02:18:39.359 OpenSearch:Enabling execution of OPENSEARCH_HOME/bin/opensearch-performance-analyzer/performance-analyzer-agent-cli for OpenSearch Performance Analyzer Plugin
02:18:40.838 OpenSearch:Error: Could not find or load main class Cannot
02:18:40.838 OpenSearch:Caused by: java.lang.ClassNotFoundException: Cannot
Error:  DOCKER> [docker.io/opensearchproject/opensearch:2.9.0] "opensearch": Container stopped with exit code 1 unexpectedly after 2015 ms while waiting on url http://localhost:9200
Error:  DOCKER> Error occurred during container startup, shutting down...
[INFO] DOCKER> [docker.io/opensearchproject/opensearch:2.9.0] "opensearch": Stop and removed container 4fdd4fbefe87 after 0 ms
Error:  DOCKER> I/O Error [[docker.io/opensearchproject/opensearch:2.9.0] "opensearch": Container stopped with exit code 1 unexpectedly after 2015 ms while waiting on url http://localhost:9200]
[INFO] 
```

**Comment by zakkak** [comment]
Build failure due to https://github.com/graalvm/mandrel/issues/665

**Comment by zakkak** [comment]
Still https://github.com/graalvm/mandrel/issues/665

**Comment by jerboaa** [comment]
Wow everything passing. We have this... every once in a blue moon ;-)

**Comment by jerboaa** [comment]
* Q IT Spring and  Q IT Misc4 issues above look to be https://github.com/quarkusio/quarkus/issues/37657 (upstream issue: https://github.com/oracle/graal/issues/8075) which is supposed to be fixed. Maybe it isn't.

**Comment by zakkak** [comment]
Caused by https://github.com/graalvm/mandrel/issues/676

**Comment by jerboaa** [comment]
Caused by https://github.com/graalvm/mandrel/issues/676

**Comment by jerboaa** [comment]
> Unfortunately, the build failed!
> 
>     * [Q main M 23 latest / Q IT Security3](https://github.com/graalvm/mandrel/actions/runs/7823563745/job/21345326029)
>       
>       * Step: Build with Maven
>         Filtered Logs:
> 
> 
> ```
> 2024-02-08T02:30:54.6653159Z [INFO] Quarkus - Integration Tests - Keycloak Policy Enforcer FAILURE [09:40 min]
> ```
> 
> Link to failing CI run: https://github.com/graalvm/mandrel/actions/runs/7823563745

Looks infra related:

```
Caused by: org.testcontainers.containers.ContainerFetchException: Can't get Docker image: RemoteDockerImage(imageName=quay.io/keycloak/keycloak:23.0.4
```

**Comment by zakkak** [comment]
Failures tracked in https://github.com/quarkusio/quarkus/issues/38851

**Comment by zakkak** [comment]
Tracked in https://github.com/graalvm/mandrel/issues/681

**Comment by jerboaa** [comment]
Reopening as this has been **wrongly** quiet the past month due to #687.

Adding the current - as of March 22, 2024 - summary of nightly build/test issues here:

- `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634. Link to [CI failure](https://github.com/graalvm/mandrel/actions/runs/8384081617/job/22961614228#step:12:1631).
- `Amazon Lambda AWS Gateway REST API` failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506. Also seen for `quarkus-integration-test-liquibase-mongodb` in IT Data 3 and `quarkus-integration-test-resteasy-jackson` in IT HTTP. Example [CI failure link](https://github.com/graalvm/mandrel/actions/runs/8384081617/job/22961614119#step:12:676).
- `Integration Tests - Locales - Some` failing due to: https://github.com/quarkusio/quarkus/issues/39478 Link to [CI failure here](https://github.com/graalvm/mandrel/actions/runs/8384081617/job/22961705532#step:12:2499).

**Comment by jerboaa** [comment]
Current CI failures are:

* Q IT Messaging1: `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634). Link to [CI failure](https://github.com/graalvm/mandrel/actions/runs/8384081617/job/22961614228#step:12:1631).
* Q IT Security1 failure of `Integration Tests - Elytron Undertow` and others failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: [[JDK 23+14] Running quarkus with latest early access build fails setting the context class loader quarkusio/quarkus#39526 (comment)](https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506). [CI failure link](https://github.com/graalvm/mandrel/actions/runs/8413746359/job/23037330043#step:12:1177).
* `Integration Tests - Locales - Some` failing due to: [[GraalVM 24.1] Integration Tests - Locales - Some fails with: Error occurred during initialization of boot layer quarkusio/quarkus#39478](https://github.com/quarkusio/quarkus/issues/39478) Link to [CI failure here](https://github.com/graalvm/mandrel/actions/runs/8413746359/job/23037331107#step:12:2458).
* Q IT Data6: `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
Current CI failures are:

* Q IT Messaging1: `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634). Link to [CI failure](https://github.com/graalvm/mandrel/actions/runs/8384081617/job/22961614228#step:12:1631).
* [Q main M 23 latest / Q IT Data3](https://github.com/graalvm/mandrel/actions/runs/8423245231/job/23087287121) and some others failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: [[JDK 23+14] Running quarkus with latest early access build fails setting the context class loader quarkusio/quarkus#39526 (comment)](https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506). [CI failure link](https://github.com/graalvm/mandrel/actions/runs/8413746359/job/23037330043#step:12:1177).
* `Integration Tests - Locales - Some` failing due to: [[GraalVM 24.1] Integration Tests - Locales - Some fails with: Error occurred during initialization of boot layer quarkusio/quarkus#39478](https://github.com/quarkusio/quarkus/issues/39478) Link to [CI failure here](https://github.com/graalvm/mandrel/actions/runs/8413746359/job/23037331107#step:12:2458).
* Q IT Data6: `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by zakkak** [comment]
This is not actually fixed, it's caused by a Quarkus build failure.

**Comment by zakkak** [comment]
This issue persists although I couldn't reproduce it locally

**Comment by jerboaa** [comment]
> This issue persists although I couldn't reproduce it locally

It should be fixed with the next run. The fix should be this one:
https://github.com/quarkusio/quarkus/commit/7331c416ebf5a4de224e2507d7f871a0cc997caf

The built revision 72c7f66a7170211c2cc6890d82da5b1313d7b523 doesn't seem to include that.

**Comment by jerboaa** [comment]
Current CI failures are:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181035283): `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634).
* [Q main M 23 latest / Q IT HTTP](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181036228) and some others failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: [[JDK 23+14] Running quarkus with latest early access build fails setting the context class loader quarkusio/quarkus#39526 (comment)](https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506).
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181034815): `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
Still same 3 failure cases:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181035283): `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634).
* [Q main M 23 latest / Q IT HTTP](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181036228) and some others failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: [[JDK 23+14] Running quarkus with latest early access build fails setting the context class loader quarkusio/quarkus#39526 (comment)](https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506).
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/8461056667/job/23181034815): `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
Still same 3 failure cases:

* Q main M 23 latest / Q IT Messaging1: `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634).
* [Q main M 23 latest / Q IT Data4] and some others failing due the `CCL.setClassLoader()` access issue caused by https://bugs.openjdk.org/browse/JDK-8328366. See the comment on the relevant issue: [[JDK 23+14] Running quarkus with latest early access build fails setting the context class loader quarkusio/quarkus#39526 (comment)](https://github.com/quarkusio/quarkus/issues/39526#issuecomment-2014799506).
* Q main M 23 latest / Q IT Data6: `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by zakkak** [comment]
Data1 and Data2 failures are Docker related, e.g.:

```
Error: Unable to start container id [9c4780bb7a02] : {"message":"driver failed programming external connectivity on endpoint mariadb-1
```

Let's see if they will reappear.

The rest are the same as in https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526

**Comment by jerboaa** [comment]
Failures still the 3 known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2031529526 (`jdk-23+18` should be coming real soon now).

**Comment by jerboaa** [comment]
The `CCL.setClassLoader()` issues are now gone since latest builds are with `jdk-23+18` or better.

The other two issues remain the same:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/8682934417/job/23808791094): `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to the following quarkus/kafka issue: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634).
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/8682934417/job/23808790649): `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
These are the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/8682934417/job/23808791094): `quarkus-integration-test-kafka-oauth-keycloak` native integration failure due to the following quarkus/kafka issue: [[JDK 23+15] org.apache.kafka.common.security.oauthbearer.internals.OAuthBearerSaslClientCallbackHandler.handleCallback fails with UnsupportedOperationException quarkusio/quarkus#39634](https://github.com/quarkusio/quarkus/issues/39634).
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/8682934417/job/23808790649): `quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling` fails due to a version parsing issue of `23-beta` as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on `22-beta` builds.

**Comment by jerboaa** [comment]
Still the two known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2058569846

**Comment by jerboaa** [comment]
Still the two known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2058569846

**Comment by zakkak** [comment]
Still the two known issues. See https://github.com/graalvm/mandrel/issues/652#issuecomment-2058569846

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
Also the Quarkus build fails with:

```
 Error:  Failed to execute goal on project quarkus-integration-test-webjars-locator: Could not resolve dependencies for project io.quarkus:quarkus-integration-test-webjars-locator:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.quarkus:quarkus-webjars-locator:jar:999-SNAPSHOT (absent), io.quarkus:quarkus-webjars-locator-deployment:pom:999-SNAPSHOT (absent): Could not find artifact io.quarkus:quarkus-webjars-locator:jar:999-SNAPSHOT -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project quarkus-integration-test-webjars-locator: Could not resolve dependencies for project io.quarkus:quarkus-integration-test-webjars-locator:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.quarkus:quarkus-webjars-locator:jar:999-SNAPSHOT (absent), io.quarkus:quarkus-webjars-locator-deployment:pom:999-SNAPSHOT (absent): Could not find artifact io.quarkus:quarkus-webjars-locator:jar:999-SNAPSHOT
```

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
The mandrel build fails. Tracked here: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
Still: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
Still: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
Still: https://github.com/graalvm/mandrel/issues/717

**Comment by jerboaa** [comment]
Still: https://github.com/graalvm/mandrel/issues/717

**Comment by zakkak** [comment]
Still:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by zakkak** [comment]
@jerboaa given that these two have been failing for quite some time (and the Kafka one is not expected to get fixed soon) what do you think would be the best way to move forward? Would it make sense to ignore the failures for the time being?

**Comment by jerboaa** [comment]
> @jerboaa given that these two have been failing for quite some time (and the Kafka one is not expected to get fixed soon) what do you think would be the best way to move forward? Would it make sense to ignore the failures for the time being?

Good question. It would be nice to have a way of ignoring test failures with a sort-of allow list file, which references the issue why it's in that allow-list file. So if you have the cycles to implement something like that it would be fantastic.

**Comment by jerboaa** [comment]
Back to the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by zakkak** [comment]
Still:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

The DevTools failure seems infra related. I can't even see the logs for some of the steps in the job. The raw logs are:

```
2024-05-15T02:13:01.6465712Z With the provided path, there will be 1 file uploaded
2024-05-15T02:13:01.6471357Z Artifact name is valid!
2024-05-15T02:13:01.6473790Z Root directory input is valid!
2024-05-15T02:13:06.6811358Z Attempt 1 of 5 failed with error: Request timeout: /twirp/github.actions.results.api.v1.ArtifactService/CreateArtifact. Retrying request in 3000 ms...
2024-05-15T02:13:14.6947963Z Attempt 2 of 5 failed with error: Request timeout: /twirp/github.actions.results.api.v1.ArtifactService/CreateArtifact. Retrying request in 6666 ms...
2024-05-15T02:13:26.3699193Z Attempt 3 of 5 failed with error: Request timeout: /twirp/github.actions.results.api.v1.ArtifactService/CreateArtifact. Retrying request in 8933 ms...
2024-05-15T02:13:40.3190268Z Attempt 4 of 5 failed with error: Request timeout: /twirp/github.actions.results.api.v1.ArtifactService/CreateArtifact. Retrying request in 12115 ms...
2024-05-15T02:13:57.4464027Z ##[error]Failed to CreateArtifact: Failed to make request after 5 attempts: Request timeout: /twirp/github.actions.results.api.v1.ArtifactService/CreateArtifact
```

**Comment by jerboaa** [comment]
Back to the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by zakkak** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

The windows failure seems infrastructure related:

```
throw new Error('Upload progress stalled.');
```

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by jerboaa** [comment]
Still the two known issues:

* [Q main M 23 latest / Q IT Messaging1](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223712): quarkus-integration-test-kafka-oauth-keycloak native integration failure due to: https://github.com/quarkusio/quarkus/issues/39634.
* [Q main M 23 latest / Q IT Data6](https://github.com/graalvm/mandrel/actions/runs/9055898848/job/24878223386): quarkus-integration-test-hibernate-search-orm-elasticsearch-outbox-polling fails due to a version parsing issue of 23-beta as reported in https://github.com/quarkusio/quarkus/issues/39664 (caused by this upstream awaitability issue: https://github.com/awaitility/awaitility/issues/277). We see the same on 22-beta builds.

**Comment by zakkak** [comment]
Superseded by #742
